File: 2144-minimum-cost-of-buying-candies-with-discount/2144-minimum-cost-of-buying-candies-with-discount.cpp
class Solution {
public:
    int minimumCost(vector<int>& cost) {
        int minCost = 0, n = cost.size(), mult = 0;
        sort(cost.begin(), cost.end());
        for(int i = n - 1; i >= 0; --i) {
            if(mult == 2) mult = 0;
            else if(mult < 2) {
                minCost += cost[i];
                ++mult;
            }
        }
        return minCost;
    }
};

File: 2144-minimum-cost-of-buying-candies-with-discount/2144-minimum-cost-of-buying-candies-with-discount.py
class Solution:
    def minimumCost(self, cost: List[int]) -> int:
        minCost, n, mult = 0, len(cost), 0
        cost = sorted(cost)
        for i in range (n - 1, -1, -1):
            if mult == 2: mult = 0
            elif mult < 2:
                minCost += cost[i]
                mult += 1
        return minCost